src/providers/DefinitionProvider.ts

```diff
@@ -85,99 +114,46 @@ export class SystemVerilogDefinitionProvider implements DefinitionProvider {
-                    const container = moduleFromPort(document, range);
-                    if (container) {
-                        await commands
-                            .executeCommand('vscode.executeWorkspaceSymbolProvider', `¤${container}`)
-                            .then((res: SymbolInformation[]) =>
-                                Promise.all(
-                                    res.map(async (x) =>
-                                        commands
-                                            .executeCommand(
-                                                'vscode.executeDocumentSymbolProvider',
-                                                x.location.uri,
-                                                word
-                                            )
-                                            .then((symbols) => {
-                                                getDocumentSymbols(
-                                                    results,
-                                                    symbols,
-                                                    word,
-                                                    range,
-                                                    x.location.uri,
-                                                    container
-                                                );
-                                            })
-                                    )
-                                )
-                            );
-                    }
-                }
-            }
         });
     }
 }
 
-// Retrieves locations from the hierarchical DocumentSymbols
-function getDocumentSymbols(
-    results: Location[],
-    entries,
+function flattenDocumentSymbols(docSyms) {
+    let o = [];
+    if (docSyms) {
+        for (let s of docSyms) {
+            o.push(s);
+            o = o.concat(flattenDocumentSymbols(s.children));
+        }
+    }
+    return o;
+}
+
+function extractLocations(
+    docSyms: Array<DocumentSymbol | SymbolInformation>,
     word: string,
-    range: Range,
     uri: Uri,
     containerName?: string
-): void {
-    if (!entries) {
-        return;
-    }
-    for (const entry of entries) {
-        if (entry.name === word && entry.kind !== getSymbolKind('potential_reference')) {
+): Location[] {
+    let locs: Location[] = [];
+    for (let s of flattenDocumentSymbols(docSyms)) {
+        if (s.name === word && s.kind !== getSymbolKind('potential_reference')) {
             if (containerName) {
-                if (entry.containerName === containerName) {
-                    results.push({
+                if (s.containerName === containerName) {
+                    locs.push({
                         uri,
-                        range: entry.range
+                        range: s.range
                     });
                 }
-            } else if (range.start.line !== entry.range.start.line) {
```

Comment:
I was thinking it made sense to provide itself as definition if it is indeed the only definition.
Not a big difference, and it didn't look like it impacted the ReferenceProvider.
I don't really have strong feelings on this, but showing that it found itself has been a bit nice while testing at least.

Reply:
I guess what we can do is to both look in the current file and in the index to look for definitions regardless of if it finds it in the current file.
(Now it will stop once it finds a definition in current document).
It's done this way as a performance/time optimisation, however thinking of it, maybe the cost of looking through the index isn't to bad.

Reply:
OK, let me play around with it in one of my large workspaces to see if there are any issues with this.

Reply:
@eirikpre I did some testing and I don't see any issue with the new functionality introduced in this branch.